comp/collector/collector/collectorimpl/agent_check_metadata.go

```diff
@@ -0,0 +1,140 @@
+	hostMetadataUtils "github.com/DataDog/datadog-agent/comp/metadata/host/hostimpl/utils"
+	"github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/collector/externalhost"
+	"github.com/DataDog/datadog-agent/pkg/collector/runner/expvars"
+	"github.com/DataDog/datadog-agent/pkg/serializer/marshaler"
+	jmxStatus "github.com/DataDog/datadog-agent/pkg/status/jmx"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+//
+// The "agent_check" metadata payload contains information about all running checks and the additional hostnames they
+// added to the Agent.
+//
+
+const (
+	defaultInterval   = 10 * time.Minute
+	firstPayloadDelay = 1 * time.Minute
+)
+
+// Payload handles the JSON unmarshalling of the metadata payload
+type Payload struct {
+	hostMetadataUtils.CommonPayload
+	Meta             hostMetadataUtils.Meta `json:"meta"`
+	AgentChecks      []interface{}          `json:"agent_checks"`
+	ExternalhostTags externalhost.Payload   `json:"external_host_tags"`
+}
+
+// MarshalJSON serialization a Payload to JSON
+func (p *Payload) MarshalJSON() ([]byte, error) {
+	// use an alias to avoid infinite recursion while serializing
+	type PayloadAlias Payload
+
+	return json.Marshal((*PayloadAlias)(p))
+}
+
+// SplitPayload breaks the payload into times number of pieces
+func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
+	return nil, fmt.Errorf("AgentChecks Payload splitting is not implemented")
+}
+
+// GetPayload builds a payload of all the agentchecks metadata
+func (c *collectorImpl) GetPayload(ctx context.Context) *Payload {
+	hostname, _ := hostname.Get(ctx)
+
+	meta := hostMetadataUtils.GetMetaFromCache(ctx, c.config)
+	meta.Hostname = hostname
+
+	cp := hostMetadataUtils.GetCommonPayload(hostname, c.config)
+	payload := &Payload{
+		CommonPayload:    *cp,
+		Meta:             *meta,
+		ExternalhostTags: *externalhost.GetPayload(),
+	}
+
+	checkStats := expvars.GetCheckStats()
+	for _, stats := range checkStats {
+		for _, s := range stats {
+			var status []interface{}
+			if s.LastError != "" {
+				status = []interface{}{
+					s.CheckName, s.CheckName, s.CheckID, "ERROR", s.LastError, "",
+				}
+			} else if len(s.LastWarnings) != 0 {
+				status = []interface{}{
+					s.CheckName, s.CheckName, s.CheckID, "WARNING", s.LastWarnings, "",
+				}
+			} else {
+				status = []interface{}{
+					s.CheckName, s.CheckName, s.CheckID, "OK", "", "",
+				}
+			}
+			if status != nil {
+				payload.AgentChecks = append(payload.AgentChecks, status)
+			}
+		}
+	}
+
+	loaderErrors := collector.GetLoaderErrors()
+	for check, errs := range loaderErrors {
+		jsonErrs, err := json.Marshal(errs)
+		if err != nil {
+			log.Warnf("Error formatting loader error from check %s: %v", check, err)
+		}
+		status := []interface{}{
+			check, check, "initialization", "ERROR", string(jsonErrs),
```

Comment:
question maybe a bit stupid but why do we 6 elements in the list above and only 5 here ?

Reply:
No dumb questions ! This code has been moved without almost any change (not sure why GH can track the move).

Not sure why we have only 5 elem here.